File: app/src/main/java/zhan/rnadomgirdanimation/refresh/SimpleDiffCallBack.java
package zhan.rnadomgirdanimation.refresh;

/**
 * Created by zhan on 2016/12/11.
 */

public class SimpleDiffCallBack extends IDiffCallBack<Integer> {

    @Override
    public boolean isItemsTheSame(int oldItemPosition, int newItemPosition) {
        return getMOldData().get(oldItemPosition) == getMNewData().get(newItemPosition);
    }

    @Override
    public boolean isContentsTheSame(int oldItemPosition, int newItemPosition) {
        return getMOldData().get(oldItemPosition) == getMNewData().get(newItemPosition);
    }
}

File: app/src/main/java/zhan/rnadomgirdanimation/refresh/SimpleDiffCallBack.kt
package zhan.rnadomgirdanimation.refresh

class SimpleDiffCallBack : IDiffCallBack<Int>() {

    override fun isItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return mOldData!![oldItemPosition] === mNewData!![newItemPosition]
    }

    override fun isContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return mOldData!![oldItemPosition] === mNewData!![newItemPosition]
    }
}
